### PR TITLE
Added ability to control sync intervals and sync always

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/filestore.go [hunk 1]
```diff
@@ -186,42 +188,44 @@ type fileStore struct {
...
+	ctmr       *time.Timer
+	werr       error
+	dmap       avl.SequenceSet
+	fch        chan struct{}
+	qch        chan struct{}
+	lchk       [8]byte
+	loading    bool
+	flusher    bool
+	noTrack    bool
+	needSync   bool
```
Comment (neilalexander) [inline server/filestore.go:226]: AFAICT `needSync` is never not `false`?

**Comment by ripienaar** [comment]
Be good if this setting is surfaced somewhere like `JetStreamVarz.Config` fetched via varz

File: server/filestore.go (hunk 1 above)
Comment (derekcollison) [reply server/filestore.go:226]: Thanks good catch, was getting late..

Fixed in separate CL but once LGTM I will squash.

**Comment by derekcollison** [comment]
Will make sure its returned.

**Review by neilalexander — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
ok added in the jsz / varz and additional test to watch needSync flag etc.

**Review by neilalexander — APPROVED** [review]
LGTM

**Pull request merged**